testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/client/ClientPolicyBasicsTest.java

```diff
@@ -0,0 +1,938 @@
+import org.keycloak.protocol.oidc.OIDCAdvancedConfigWrapper;
+import org.keycloak.protocol.oidc.OIDCLoginProtocol;
+import org.keycloak.representations.AccessToken;
+import org.keycloak.representations.RefreshToken;
+import org.keycloak.representations.idm.ClientInitialAccessCreatePresentation;
+import org.keycloak.representations.idm.ClientInitialAccessPresentation;
+import org.keycloak.representations.idm.ClientRepresentation;
+import org.keycloak.representations.idm.ComponentRepresentation;
+import org.keycloak.representations.idm.EventRepresentation;
+import org.keycloak.representations.idm.RealmRepresentation;
+import org.keycloak.representations.oidc.OIDCClientRepresentation;
+import org.keycloak.representations.oidc.TokenMetadataRepresentation;
+import org.keycloak.services.clientpolicy.ClientPolicyException;
+import org.keycloak.services.clientpolicy.ClientPolicyProvider;
+import org.keycloak.services.clientpolicy.DefaultClientPolicyProviderFactory;
+import org.keycloak.services.clientpolicy.condition.ClientPolicyConditionProvider;
+import org.keycloak.services.clientpolicy.executor.ClientPolicyExecutorProvider;
+import org.keycloak.testsuite.AbstractKeycloakTest;
+import org.keycloak.testsuite.AssertEvents;
+import org.keycloak.testsuite.admin.ApiUtil;
+import org.keycloak.testsuite.arquillian.annotation.EnableFeature;
+import org.keycloak.testsuite.services.clientpolicy.condition.TestAuthnMethodsConditionFactory;
+import org.keycloak.testsuite.services.clientpolicy.condition.TestClientRolesConditionFactory;
+import org.keycloak.testsuite.services.clientpolicy.condition.TestRaiseExeptionConditionFactory;
+import org.keycloak.testsuite.services.clientpolicy.executor.TestClientAuthenticationExecutorFactory;
+import org.keycloak.testsuite.services.clientpolicy.executor.TestPKCEEnforceExecutorFactory;
+import org.keycloak.testsuite.util.OAuthClient;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+@EnableFeature(value = Profile.Feature.CLIENT_POLICIES, skipRestart = true)
+public class ClientPolicyBasicsTest extends AbstractKeycloakTest {
+
+    private static final Logger logger = Logger.getLogger(ClientPolicyBasicsTest.class);
+
+    static final String REALM_NAME = "test";
+    static final String TEST_CLIENT = "test-app";
+
+    ClientRegistration reg;
+
+    @Rule
+    public AssertEvents events = new AssertEvents(this);
+
+    @Before
+    public void before() throws Exception {
+        // get initial access token for Dynamic Client Registration with authentication
+        reg = ClientRegistration.create().url(suiteContext.getAuthServerInfo().getContextRoot() + "/auth", REALM_NAME).build();
+        ClientInitialAccessPresentation token = adminClient.realm(REALM_NAME).clientInitialAccess().create(new ClientInitialAccessCreatePresentation(0, 10));
+        reg.auth(Auth.token(token));
+
+        System.setProperty("keycloak.profile", "preview");
```

Comment:
Is this needed in the pull request? Some other tests, which rely on profile features, don't need to set this system property explicitly?

Reply:
Yes, I'll remove `testPurgePreviewProfile` and its related codes as you've pointed out. This test was for checking activating/deactivating preview feature.

Reply:
Cool, Thanks. If tests still work for you with:
```
mvn clean install -DskipTests=true -Pdistribution
```
and then both Keycloak on undertow and wildfly
````
mvn -f testsuite/integration-arquillian/pom.xml -Dtest=ClientPolicyBasicsTest clean install
mvn -f testsuite/integration-arquillian/pom.xml -Pauth-server-wildfly -Dtest=ClientPolicyBasicsTest clean install
````
then we should be fine.